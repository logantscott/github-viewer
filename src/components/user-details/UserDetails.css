.UserDetails{
  a {
    color: blue;
    text-decoration: none;
    font-weight: bold;
    font-size: 2rem;
  },
  a:hover, a:active {
    text-decoration: underline;
  },
  h2 {
    margin: 0;
  },
  img {
    width: 12rem;
    margin: 0.5rem 0;
  },
  p {
    margin: 0.3rem 0;
  }
}
