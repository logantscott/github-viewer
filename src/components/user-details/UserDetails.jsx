import React from 'react';
import PropTypes from 'prop-types';
import styles from './UserDetails.css';

const UserDetails = ({ userDetails: { login, avatar_url, url, followers, following } }) => (
  <div className={styles.UserDetails}>
    <a href={url}><img src={avatar_url} alt={login} /></a>
    <a href={url}><h2>{login}</h2></a>
    <p>Followers: {followers}, Following: {following}</p>
  </div>
);

UserDetails.propTypes = {
  userDetails: (
    PropTypes.object.isRequired,
    PropTypes.shape({
      login: PropTypes.string.isRequired,
      avatar_url: PropTypes.string,
      url: PropTypes.string.isRequired,
      followers: PropTypes.number,
      following: PropTypes.number
    })
  )
};

export default UserDetails;
