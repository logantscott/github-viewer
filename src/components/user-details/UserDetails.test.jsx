import React from 'react';
import { shallow } from 'enzyme';
import UserDetails from './UserDetails';

const userDetails = {
  login: 'logantscott',
  url: 'https://github.com/logantscott',
  avatar_url: 'https://avatars2.githubusercontent.com/u/57209519?s=460&u=8fad17c44cbb8ea89b7fc34c8475dd4d181a9604&v=4',
  followers: 1,
  following: 2
};

describe('UserDetails component', () => {
  it('renders UserDetails', () => {
    const wrapper = shallow(<UserDetails userDetails={userDetails} />);
    expect(wrapper).toMatchSnapshot();
  });
});
  