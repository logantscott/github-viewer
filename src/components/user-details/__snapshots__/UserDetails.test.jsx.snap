// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetails component renders UserDetails 1`] = `
<div
  className="UserDetails"
>
  <a
    href="https://github.com/logantscott"
  >
    <img
      alt="logantscott"
      src="https://avatars2.githubusercontent.com/u/57209519?s=460&u=8fad17c44cbb8ea89b7fc34c8475dd4d181a9604&v=4"
    />
  </a>
  <a
    href="https://github.com/logantscott"
  >
    <h2>
      logantscott
    </h2>
  </a>
  <p>
    Followers: 
    1
    , Following: 
    2
  </p>
</div>
`;
